backend/TreeOfAKind.Infrastructure/FileStorage/AzureBlobStorageService.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using MimeTypes;
+using Serilog;
+using TreeOfAKind.Application.Services;
+using TreeOfAKind.Domain.Trees;
+
+namespace TreeOfAKind.Infrastructure.FileStorage
+{
+    public class AzureBlobStorageService : IFileSaver
+    {
+        private readonly ILogger _logger;
+        private readonly IDictionary<string, string> _metadata;
+        private readonly BlobServiceClient _blobServiceClient;
+
+        private const string Wildcard = "*";
+
+
+        private static readonly BlobRequestConditions BlobRequestConditions
+            = new BlobRequestConditions {IfNoneMatch = new ETag(Wildcard)};
+
+        public AzureBlobStorageService(AzureBlobStorageSettings settings, ILogger logger)
+        {
+            _logger = logger;
+            var connectionString = settings.ConnectionString;
+            _metadata = settings.Metadata;
+
+            _blobServiceClient = new BlobServiceClient(connectionString);
+        }
+
+        public async Task<Uri> UploadFile(string containerName, string contentType, Stream stream,
+            CancellationToken cancellationToken = default!)
+        {
+            try
+            {
+                var headers = new BlobHttpHeaders {ContentType = contentType};
+                var blobName = Guid.NewGuid() + MimeTypeMap.GetExtension(contentType);
+
+                var blobContainerClient = _blobServiceClient.GetBlobContainerClient(containerName);
+                await blobContainerClient.CreateIfNotExistsAsync(PublicAccessType.Blob, _metadata , cancellationToken);
```

Comment:
We probably don't want public access containers/blobs, these things are private.

Reply:
Containers are private, u cannot list the content of a container, but u can download a specific file if u have full path